src/mscorlib/src/System/TimeZoneInfo.cs

```diff
@@ -3816,7 +4142,7 @@ private static string GetTimeZoneDirectory()
             // skip ahead of the Leap-Seconds Adjustment data.  In a future release, consider adding
             // support for Leap-Seconds
             //
-            index += (int)(t.LeapCount * 8); // skip the leap second transition times
+            index += (int)(t.LeapCount * (timeValuesLength + 4)); // skip the leap second transition times
```

Comment:
From 'man tzfile':
>Then  there are tzh_leapcnt pairs of four-byte values, written in standard byte order; the first value  of  each  pair  gives  the  time  (as returned  by  time(2))  at which a leap second occurs; **the second gives the total number of leap seconds to be applied after  the  given  time**.  The pairs of values are sorted in ascending order by time.

This is the definition for Version 1 tzfiles.  For Version 2:
>  For  version-2-format timezone files, the above header and data is followed by a second header and data,  identical  in  format  except  that **eight  bytes  are  used  for  each transition time or leap-second time**.

So the 4 here is the 4 bytes to tell how many leap seconds have occurred after this point in time.  It should always be 4, even if sizeof(int) would happen to change.